macro-mathjax-ui/src/main/resources/Macros/MathJaxMacro.xml

```diff
@@ -149,8 +149,27 @@ y = \frac{c}{cb-ad}
     startup: {
       pageReady() {
         document.body.classList.add('mathjax-ignore');
-        return MathJax.startup.defaultPageReady();
-      }
+        // Since the same macro will be displayed twice in a rendered diff, we typeset each one individually in order to
+        // avoid errors from duplicate labels. This will produce some inconsistences with the displayed equation
+        // numbers, meaning that each macro will start counting from 1 and anchor ids will be duplicated.
+        if ($('#renderedChanges').length &gt; 0) {
```

Comment:
I assume this code handles the typesetting for the entire web page, right? What if we have a MathJax macro call in a panel or anywhere **outside** the ``#renderedChanges`` container (e.g. in a UI extension). It feels strange that the presence / absence of ``#renderedChanges`` element influences the rendering for the **entire** page. I would have expected that the typesetting **inside** ``#renderedChanges`` is different, but everything outside is not affected. Maybe this is not possible?

Reply:
@mflorea Indeed, I had only considered the base case. I created a new PR to address  this problem https://github.com/xwiki-contrib/macro-mathjax/pull/4
With this, after the `#renderedChanges` math elements are typeset, I let MathJax handle the other math elements on the page, without considering the already typeset elements in order to avoid errors.
Also, along with of this, I had to handle the case when `#renderedChanges` content appears after MathJax is loaded (by other math elements on the page)